presto-hive/src/main/java/com/facebook/presto/hive/metastore/Storage.java

```diff
@@ -77,6 +79,42 @@ public boolean isSkewed()
         return serdeParameters;
     }
 
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .add("skewed", skewed)
+                .add("storageFormat", storageFormat)
+                .add("location", location)
+                .add("bucketProperty", bucketProperty)
+                .add("serdeParameters", serdeParameters)
+                .toString();
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        Storage storage = (Storage) o;
+        return skewed == storage.skewed &&
+                Objects.equals(storageFormat, storage.storageFormat) &&
+                Objects.equals(location, storage.location) &&
+                Objects.equals(bucketProperty, storage.bucketProperty) &&
+                Objects.equals(serdeParameters, storage.serdeParameters);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(storageFormat, location, bucketProperty, skewed, serdeParameters);
```

Comment:
can you have the fields in the same order here as in toString and equals?
ie `skewed` is first there